cmd/notification.go

```diff
@@ -1528,7 +1529,24 @@ func (sys *NotificationSys) GetReplicationMRF(ctx context.Context, bucket, node
 	}(mrfCh)
 	go func(wg *sync.WaitGroup) {
 		wg.Wait()
-		defer close(mrfCh)
+		CloseChannelWithoutEmpty(ctx, mrfCh)
 	}(&wg)
 	return mrfCh, nil
 }
+
+// CloseChannelWithoutEmpty - close channel without empty
+func CloseChannelWithoutEmpty[T any](ctx context.Context, ch chan T) {
+	defer close(ch)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			if len(ch) == 0 {
+				return
+			}
+			// make sure all the data have being sent
+			runtime.Gosched()
+		}
+	}
+}
```

Comment:
why do we need to do this? this will create a busy loop from the `for { select { default: ` there is no delay there so it will shoot up CPU 

Can you explain what you are trying to achieve?

Reply:
Because wg.Wait() ensures that all the entries in channel are empty before close()